Fix: Check for debug mode when watching cutscene debug inputs

Archez opened the pull request:
There are three inputs that can interrupt/restart cutscenes from the debug rom. The interrupt code flow has a check for "debug mode", but it appears the restart code flow did not have the same check. There are some vanilla cutscenes that when playing/finished, if pressing DUP or DLEFT, the debug restart cutscene flow would process.

This PR adds "debug mode" checks to the DUP and DLEFT code paths.

Fixes #3129, fixes #1677

<!--- section:artifacts:start -->
### Build Artifacts
  - [soh.otr.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/892254758.zip)
  - [soh-linux-compatibility.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/892254760.zip)
  - [soh-linux-performance.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/892254761.zip)
  - [soh-mac.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/892254762.zip)
  - [soh-switch.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/892254764.zip)
  - [soh-wiiu.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/892254765.zip)
  - [soh-windows.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/892254767.zip)
<!--- section:artifacts:end -->

Archez committed:
fix check for debug mode when watching cutscene debug inputs

garrettjoecox commented:
So just to be clear, this is real behavior on the debug rom? If so I think we either need to cvar it (defaulted to on) or at least document why we changed it

Archez commented:
> So just to be clear, this is real behavior on the debug rom? If so I think we either need to cvar it (defaulted to on) or at least document why we changed it

Decomp is based on the debug rom, so anything debug is going to be baked into the source like this.

In SoH we had to add the debug mode CVar checks to all these debug code paths to prevent them from running normally. There was already checks for DRIGHT and START actions for cutscenes, but we were missing the same checks for DLEFT and DUP.

https://github.com/HarbourMasters/Shipwright/blob/3fb698bab87fc7d4687cdd76b9c2fdd5bb0d4420/soh/src/code/z_demo.c#L501-L502

https://github.com/HarbourMasters/Shipwright/blob/3fb698bab87fc7d4687cdd76b9c2fdd5bb0d4420/soh/src/code/z_demo.c#L1646-L1649

We wouldn't want these actions to run for normal game play if debug mode is off, hence the CVar checks I added to match everywhere else.

garrettjoecox reviewed (approved):
> In SoH we had to add the debug mode CVar checks to all these debug code paths to prevent than from running normally.

Ah TIL, I guess that makes sense I had just never thought about it. LGTM

jbodner09 commented:
In #454 I used the gDebugCamera cvar instead of just debug mode.  Any chance you want to change that cvar over to just the debug mode one while you're at it?

garrettjoecox commented:
> In #454 I used the gDebugCamera cvar instead of just debug mode. Any chance you want to change that cvar over to just the debug mode one while you're at it?

@Archez thoughts here? This PR is merge-able otherwise

Archez committed:
update other cvar to debug enabled

Archez commented:
Updated.
I think it may be worth considering having extra debug toggles so that you can enable only certain features. e.g. Debug saves with debug warp, but no cutscene/camera controls. That can be saved for a future PR if others agree.
